core/src/main/scala/org/bitcoins/core/crypto/TransactionSignatureSerializer.scala

```diff
@@ -47,32 +35,52 @@ sealed abstract class TransactionSignatureSerializer {
     */
   def serializeForSignature(
       txSigComponent: TxSigComponent,
-      hashType: HashType): ByteVector = {
+      hashType: HashType,
+      taprootOptions: TaprootSerializationOptions): ByteVector = {
     val spendingTransaction = txSigComponent.transaction
     val inputIndex = txSigComponent.inputIndex
     val output = txSigComponent.output
     val script = BitcoinScriptUtil.calculateScriptForSigning(
       txSigComponent,
       output.scriptPubKey.asm)
-    val amount = output.value
 
-    serializeForSignature(spendingTransaction,
-                          inputIndex,
-                          hashType,
-                          amount,
-                          script,
-                          txSigComponent.sigVersion)
+    txSigComponent match {
+      case _: BaseTxSigComponent | _: WitnessTxSigComponentRaw |
+          _: WitnessTxSigComponentRebuilt =>
+        serializeForSignature(spendingTransaction,
+                              inputIndex,
+                              hashType,
+                              Vector(output),
+                              script,
+                              txSigComponent.sigVersion,
+                              taprootOptions)
+      case t: TaprootTxSigComponent =>
+        serializeForSignature(spendingTransaction,
+                              inputIndex,
+                              hashType,
+                              t.outputs,
+                              script,
+                              txSigComponent.sigVersion,
+                              taprootOptions)
+    }
+
   }
 
   def serializeForSignature(
       spendingTransaction: Transaction,
       inputIndex: UInt32,
       hashType: HashType,
-      amount: CurrencyUnit,
+      outputs: Vector[TransactionOutput],
       script: Seq[ScriptToken],
-      sigVersion: SignatureVersion): ByteVector = {
+      sigVersion: SignatureVersion,
+      taprootOptions: TaprootSerializationOptions): ByteVector = {
+    val keyVersion: Byte = 0.toByte
```

Comment:
Shouldn't we get this from the pubkey / a param?

Reply:
Maybe in the future, but not now:

https://github.com/bitcoin/bitcoin/blob/a085a554913ae8f4ed83afac830ce6dc39c9cc65/src/script/interpreter.cpp#L1492